cmd/neofs-node/object.go

```diff
@@ -522,3 +537,40 @@ func (c *reputationClientConstructor) Get(info coreclient.NodeInfo) (coreclient.
 
 	return cl, nil
 }
+
+type engineWithNotifications struct {
+	e  *engine.StorageEngine
+	nw notificationWriter
+	ns netmap.State
+
+	defaultTopic string
+}
+
+func (e engineWithNotifications) Put(o *objectSDK.Object) error {
+	if err := engine.Put(e.e, o); err != nil {
+		return err
```

Comment:
Other notifications are triggered by iterating over objects stored in engine. It will be consistent to produce those immediate notifications on the same basis.

But it is really good question, though. /cc @realloc

Reply:
> is it correct?

used a rule "notify only if I managed to store the object", thought that it was expected behavior

Reply:
btw, i thought that the main idea about notifications is to say that some node successfully stored the object

Reply:
The notification must be issued only for successfully stored the object. Otherwise, the actions following the notification will have no sense.